WordPress/openverse-api
# Move page and page_size query param validation into serializer

## Fixes


Related to #857 and #843 

## Description


In this PR I move scattered query parameter validation into the query serializer, where it probably belongs. This has two benefits:

1. It removes code in favour of using built-in Django Rest Framework serializer validation
2. It removes usages of the bespoke `get_api_exception`
3. It reduces instances of a loose pattern of scattered validation where parameter validation happens multiple times and deep into the runtime of a search query, rather than once at the top (where it is expected to happen and easier to find)

None of these changes change the API surface.

The lines changed is also deceptive on this PR as something like 85 of the lines added or changed were in the tests.

## Testing Instructions

Ensure CI passes. Checkout the branch, set up the local API following the README instructions and make some smoke test requests to ensure everything works as expected.

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
